python_modules/libraries/dagster-fivetran/dagster_fivetran/fivetran_event_iterator.py

```diff
@@ -0,0 +1,32 @@
+from collections import abc
+from typing import TYPE_CHECKING, Generic, Iterator, Union
+
+from dagster import AssetMaterialization, MaterializeResult
+from typing_extensions import TypeVar
+
+if TYPE_CHECKING:
+    from dagster_fivetran.resources import FivetranWorkspace
+
+
+FivetranEventType = Union[AssetMaterialization, MaterializeResult]
+T = TypeVar("T", bound=FivetranEventType)
+
+
+class FivetranEventIterator(Generic[T], abc.Iterator):
```

Comment:
suprised we're using abc.Iterator and not typing.Iterator?

Reply:
wait why is this an abstract base class? We're instantiating it right?

Reply:
That's a good call. I replicated what we already have for other integrations - [DbtEventIterator](https://github.com/dagster-io/dagster/blob/56c282cda28449827d558b269a377a4eef5af6a6/python_modules/libraries/dagster-dbt/dagster_dbt/core/dbt_event_iterator.py#L189), [DltEventIterator](https://github.com/dagster-io/dagster/blob/56c282cda28449827d558b269a377a4eef5af6a6/python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/dlt_event_iterator.py#L61) and [SlingEventIterator](https://github.com/dagster-io/dagster/blob/56c282cda28449827d558b269a377a4eef5af6a6/python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/sling_event_iterator.py#L171) are all subclassing `abc.Iterator` - but I agree that subclassing typing.Iterator makes much more sense. Will fix here and open PRs to update the other integrations.

Reply:
Updated in [0582d6f](https://github.com/dagster-io/dagster/pull/26396/commits/0582d6fe4380cce22dc578d77cc527b4ccea15e5)